docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+# Provisioning Setup and Processing
+
+## Table of Contents
+
+- [Provisioning Setup and Processing](#provisioning-setup-and-processing)
+  - [Table of Contents](#table-of-contents)
+  - [Provisioning Resources and Schemas](#provisioning-resources-and-schemas)
+    - [Creating Access Request Schemas](#creating-access-request-schemas)
+    - [Creating Credential Request Schemas](#creating-credential-request-schemas)
+    - [Schema Builder](#schema-builder)
+    - [Schema Property Builder](#schema-property-builder)
+      - [String Property Builder](#string-property-builder)
+      - [Number Property Builder](#number-property-builder)
+      - [Integer Property Builder](#integer-property-builder)
+      - [Array Property Builder](#array-property-builder)
+  - [Handle Provisioning Requests](#handle-provisioning-requests)
+    - [Creating and Registering Provisioner](#creating-and-registering-provisioner)
+      - [RequestStatus return structure](#requeststatus-return-structure)
+    - [Application Request Provision and Deprovision](#application-request-provision-and-deprovision)
+    - [Access Request Provision and Deprovision](#access-request-provision-and-deprovision)
+    - [Credential Request Provision and Deprovision](#credential-request-provision-and-deprovision)
+
+Marketplace Provisioning allows the consumer to create applications, associate products to the application, and create credentials in the application. Amplify Agents SDK publishes API server resources which result may be added to products within the Marketplace. The agents can configure data that is required from consumers, at the time of access request, to provision access to the resource in the connected gateway.
+
+## Provisioning Resources and Schemas
+
+### Creating Access Request Schemas
+
+An Access Request Definition tells Marketplace the information that a consumer needs to provide in order for the agent to provision that consumer the ability to use a discovered API.
+
+The Amplify Agent SDK provides a mechanism to set up access request definition schemas based on configuration parameters required by the connected gateway. The agent can set up schemas which register resources of type AccessRequestDefinition in Amplify Central. When the agent discovers an API and publishes it to Amplify Central, the created resource references one of the AccessRequestDefinition resources and associates it with the published API. This could be a schema per API on the target gateway or based on the authentication type of API.
+
+The agent implementation can use the helper method `agent.NewAPIKeyAccessRequestBuilder().Register()` for an access request definition named *api-key* for API Key APIs that require no additional data for provisioning. It is also possible for the agent to create it's own access request definitions by using `agent.NewAccessRequestBuilder()` adding schema properties, see below, before calling Register.
```

Comment:
its instead of it's

add comma before ", adding"
...access request definitions by using `agent.NewAccessRequestBuilder()` , adding schema properties